### DISCUSSION TITLE
Issue with Playing .strm Files Containing Plugin URLs

### DISCUSSION THREAD

**Discussion by Spider-netizen** [PseudoTV/PseudoTV_Live#56] [Q&A]
Hi @Lunatixz ,

Thanks for your work.

Does the add-on support playing `.strm` files containing plugin url? I've managed to modify it in the past to make it work with this code:
```
if info.get('file'):
            file_path = item.get('file','')
            log(f'buildItemListItem, Translating path "{file_path}"...')
            file_path = xbmcvfs.translatePath(file_path)
            log(f'buildItemListItem, Translated path to "{file_path}".')
            listitem.setPath(file_path)
            if file_path.endswith('.strm'):
                # with open(file_path, 'r') as f:
                with io.open(file_path, mode='r', encoding="utf-8") as f:
                    file_path = f.read()
            listitem.setPath(file_path)
```

But I recently updated the add-on and I can't make this work anymore.

**Comment by Lunatixz** [comment]
sorry ignore my prior reply I misunderstood the question. Yes, PseudoTV supports strms. There is no need to edit code, simply enable "Include STRMs" in PseudoTV global settings. 

BTW STRMs contain links to media, not paths to directories. 

STRMs must also be accompanied by a nfo file matching the strms file name which is required to include duration or runtime values.

**Reply by Spider-netizen** [reply]
Hey @Lunatixz,

Thanks for the response. I'm sorry for not replying sooner, but I just wanted to make sure I'm not missing anything on my end, and I haven't come around to it till now. Just to make sure, I've updated to 0.5.4h and made sure "Include STRMs" is enabled. I still can't play `.strm` files that contain plugin URLs. 

> STRMs must also be accompanied by a nfo file matching the strms file name which is required to include duration or runtime values.

The STRMs show up in the EPG they just won't play.

**Reply by Lunatixz** [reply]
Can you post an example strm and nfo, thx.

**Reply by Spider-netizen** [reply]
> BTW STRMs contain links to media, not paths to directories.

I'm not sure what you mean by that. The STRMs I'm talking about contain plugin URLs.

The code I mentioned earlier was code I modified in "kodi.py", "buildItemListItem" method of the "Dialog" class, which seems like you've changed at some point (I'm not sure which version it was I modified).

**Reply by Spider-netizen** [reply]
> Can you post an example strm and nfo, thx.

STRM:
`plugin://plugin.video.amazon-test/?asin=amzn1.dv.gti.eedf35c9-fa32-4c9f-8516-bfa78ab759de&mode=PlayVideo&name=A+Shadow+of+the+Past&adult=0&trailer=0&selbitrate=0
`

NFO:
```
<?xml version="1.0" encoding="UTF-8" standalone="yes" ?>
<episodedetails>
<title>A Shadow of the Past</title>
<thumb>https://m.media-amazon.com/images/S/pv-target-images/8142b0f00113ee5855c8831da6a9b1e0a96a6d5f68a373b8d620acdc309dc8f2.jpg</thumb>
<aired>2022-09-02</aired>
<episode>1</episode>
<mpaa>13+</mpaa>
<plot>Series Premiere. Galadriel is disturbed by signs of an ancient evil’s return; Arondir makes an unsettling discovery; Elrond is presented with an intriguing new venture; Nori breaks the Harfoot community’s most deeply-held rule.</plot>
<aired>2022-09-02</aired>
<season>1</season>
<showtitle>The Lord of the Rings: The Rings of Power</showtitle>
<year>2022</year>
<fileinfo>
   <streamdetails>
       <video>
           <codec>h264</codec>
           <durationinseconds>3947</durationinseconds>
           <language>eng</language>
           <scantype>Progressive</scantype>
       </video>
   </streamdetails>
</fileinfo>
</episodedetails>

```

**Reply by Lunatixz** [reply]
Thx, I'm wondering why you're using strms? It's easier to create a custom PseudoTV channel in the manager with a direct path to the plugin... 

Either way, I'll test the strm you provided...

**Reply by Spider-netizen** [reply]
Thank you for the prompt replies and the help, really appreciate this. 

There are several add-ons that let you export shows and movies as STRMs, and then you can scan them into Kodi's library like any other media. That way, they show up in your library and allow you the flexibility that you would have with any other media in your library. Also, I like using PseudoTV to automatically create channels based on my library media. So, that's why I prefer this method rather than manually creating channels. I like having the STRMs show up on different channels mixed with media from different sources. I don't think I ever actually tried to create a channel with a direct path to the plugin, so I'm not even sure what it would look like.

The code I provided definitely worked as a workaround for me previously, but I'm pretty sure it broke some other stuff. 

I've been tinkering with your add-on since the very old version, so I'm a bit familiar with it, but not enough to push the features I've been working on as they do break other stuff.

**Reply by Lunatixz** [reply]
FYI Tested directly building a channel from the plugin... and it works great! I'll test STRMS next. 
Easiest method would be to navigate the plugin to a directory you'd like to make into channel then use the context menu option PseudoTV Live/Add to PseudoTV Live. Process was a lot slower then normal due to the plugins caching. Most likely because I just installed it. I'll be online for the next hour if you'd like to test now and get back to me.

**Comment by Lunatixz** [comment]
Has this issue been resolved? Is the new strm import tool useful? I'll close the thread in a few days if no new updates. THX

**Reply by Spider-netizen** [reply]
I'm not sure what the new strm import tool is, but I'll check the newest version soon and let you know if my original issue was resolved.

**Comment by quopquai** [comment]
Hi Lunatix, sorry, I think you have the wrong person. I don’t have anything
to do with a “new strm import tool”!?

quopquaoi

**Comment by Lunatixz** [comment]
Sorry for the confusion. I was asking if the new strm import function I added to PseudoTV Live a few months back is working out...
